pkg/atlas/meta.go

```diff
@@ -21,8 +31,44 @@ type InstanceSize struct {
 // GET /cloudProviders/{NAME}/options
 func (c *HTTPClient) GetProvider(name string) (*Provider, error) {
 	path := fmt.Sprintf("cloudProviders/%s/options", name)
-
 	var provider Provider
-	err := c.requestPrivate(http.MethodGet, path, nil, &provider)
+
+	// First check to see if any of the providers have been set by the administrator of the cluster locally
+	provider, isSet, err := setupProviders(name)
+	if err != nil {
+		return &provider, err
+	}
+	if isSet {
+		return &provider, err
+	}
+
+	// Otherwise we make a request to the private atlas API when it's not set
+	err = c.requestPrivate(http.MethodGet, path, nil, &provider)
 	return &provider, err
 }
+
+// setupProviders will setup all of the available providors together with their plans/sizes.
+// Users may want to set their local providor, and combine that with a remote provider that isn't set locally.
+func setupProviders(name string) (provider Provider, isSet bool, err error) {
```

Comment:
This is a bad name on the grounds that this function does not set up providers; "set up" implies some side effect when this function doesn't have any and "providers" is plural when this function takes a single provider name.